src/JSInterop/Microsoft.JSInterop/src/IJSRuntime.cs

```diff
@@ -14,21 +13,28 @@ public interface IJSRuntime
     {
         /// <summary>
         /// Invokes the specified JavaScript function asynchronously.
+        /// <para>
+        /// <see cref="JSRuntime"/> will apply timeouts to this operation based on the value configured in <see cref="JSRuntime.DefaultAsyncTimeout"/>. To dispatch a call with a different, or no timeout,
+        /// consider using <see cref="InvokeAsync{TValue}(string, CancellationToken, object[])" />.
+        /// </para>
         /// </summary>
         /// <typeparam name="TValue">The JSON-serializable return type.</typeparam>
         /// <param name="identifier">An identifier for the function to invoke. For example, the value <code>"someScope.someFunction"</code> will invoke the function <code>window.someScope.someFunction</code>.</param>
         /// <param name="args">JSON-serializable arguments.</param>
         /// <returns>An instance of <typeparamref name="TValue"/> obtained by JSON-deserializing the return value.</returns>
-        Task<TValue> InvokeAsync<TValue>(string identifier, params object[] args);
+        ValueTask<TValue> InvokeAsync<TValue>(string identifier, object[] args);
```

Comment:
Methods on the interface, including with ones with `params` are preferred over extension methods. If we declare the args array as a `params`, it results in the following method to bind to the interface rather than the extension method that accepts a TimeSpan.

```C#
var timeout = TimeSpan.FromSeconds(3);
var result = await JS.InvokeAsync<string>("someMethod", timeout, "someValue");
```

My fix for this was 

a) To make this an explicit object[]
b) Add an extension method which accepts a `params args[]` array.

Reply:
oh wow. Cool lifehack I guess? Mystery explained.